tests/large/test_rust.py

```diff
@@ -0,0 +1,72 @@
+
+# -*- coding: utf-8 -*-
+# Copyright (C) 2014 Canonical
+#
+# Authors:
+#  Didier Roche
+#  Tin Tvrtković
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation; version 3.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+"""Tests for the Rust category"""
+import subprocess
+import os
+import tempfile
+from tests.large import LargeFrameworkTests
+from tests.tools import UMAKE, spawn_process
+
+
+class RustTests(LargeFrameworkTests):
+    """The official Rust distribution"""
+
+    TIMEOUT_INSTALL_PROGRESS = 300
+
+    EXAMPLE_PROJECT = """fn main() {println!("Hello World!");}"""
+
+    def setUp(self):
+        super().setUp()
+        self.installed_path = os.path.join(self.install_base_path, "rust", "rust-lang")
+        self.framework_name_for_profile = "Rust"
+
+    @property
+    def exec_path(self):
+        return os.path.join(self.installed_path, "rustc", "bin", "rustc")
+
+    def test_default_rust_install(self):
+        """Install Rust from scratch test case"""
+        if not self.in_container:
+            self.example_prog_dir = tempfile.mkdtemp()
+            self.additional_dirs.append(self.example_prog_dir)
+            example_file = os.path.join(self.example_prog_dir, "hello.rs")
+            open(example_file, "w").write(self.EXAMPLE_PROJECT)
+            compile_command = ["bash", "-l", "-c", "rustc {}".format(example_file)]
+        else:  # our mock expects getting that path
+            compile_command = ["bash", "-l", "rustc /tmp/hello.rs"]
+
+        self.child = spawn_process(self.command('{} rustc'.format(UMAKE)))
+        self.expect_and_no_warn("Choose installation path: {}".format(self.installed_path))
+        self.child.sendline("")
+        self.expect_and_no_warn("Installation done", timeout=self.TIMEOUT_INSTALL_PROGRESS)
+        self.wait_and_close()
+
+        self.assert_exec_exists()
+        self.assertTrue(self.is_in_path(self.exec_path))
```

Comment:
As you are also adding another directory in cargo/bin, I would also assertTrue that one binary there exists + is in path to ensure both paths were added.